pennylane/ops/qubit/matrix_ops.py

```diff
@@ -211,25 +226,84 @@ def __init__(
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+        >>> qml.ControlledQubitUnitary.compute_matrix(U, control_wires=[1], wires=[0], control_values="1")
+        [[ 1.        +0.j  0.        +0.j  0.        +0.j  0.        +0.j]
+         [ 0.        +0.j  1.        +0.j  0.        +0.j  0.        +0.j]
+         [ 0.        +0.j  0.        +0.j  0.94877869+0.j  0.31594146+0.j]
+         [ 0.        +0.j  0.        +0.j -0.31594146+0.j  0.94877869+0.j]]
+        """
+        # this method reproduces the logic contained in the init function and the matrix() method
+
+        total_wires = qml.wires.Wires(control_wires) + qml.wires.Wires(wires)
+        # if control values unspecified, we control on the all-ones string
+        if not control_values:
+            control_values = "1" * len(control_wires)
+
+        control_int = int(control_values, 2)
+        padding_left = control_int * len(U)
+        padding_right = 2 ** len(total_wires) - len(U) - padding_left
+        interface = qml.math.get_interface(U)
+        left_pad = qml.math.cast_like(qml.math.eye(padding_left, like=interface), 1j)
+        right_pad = qml.math.cast_like(qml.math.eye(padding_right, like=interface), 1j)
+        return qml.math.block_diag([left_pad, U, right_pad])
+
+    def matrix(self, wire_order=None):
+        """Matrix representation of this operator.
+
+        Returns:
+            array: matrix representation
+
+        **Example**
+
+        >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+        >>> op = qml.ControlledQubitUnitary(U, control_wires=[1], wires=[0], control_values="1")
+        >>> op.matrix(wire_order=[0, 1])
+        [[ 1.        +0.j  0.        +0.j  0.        +0.j  0.        +0.j]
+         [ 0.        +0.j  0.94877869+0.j  0.        +0.j  0.31594146+0.j]
+         [ 0.        +0.j  0.        +0.j  1.        +0.j  0.        +0.j]
+         [ 0.        +0.j -0.31594146+0.j  0.        +0.j  0.94877869+0.j]]
+        """
+        if self._cu is None:
             interface = qml.math.get_interface(self.U)
             left_pad = qml.math.cast_like(qml.math.eye(self._padding_left, like=interface), 1j)
             right_pad = qml.math.cast_like(qml.math.eye(self._padding_right, like=interface), 1j)
-            self._CU = qml.math.block_diag([left_pad, self.U, right_pad])
+            self._cu = qml.math.block_diag([left_pad, self.U, right_pad])
+
+        # we do not call compute_matrix here to avoid replicating logic already computed before
+        canonical_matrix = self._cu
 
-        params = list(params)
-        params[0] = self._CU
-        return super()._matrix(*params)
+        if wire_order is None or self.wires == Wires(wire_order):
+            return canonical_matrix
+
+        return qml.operation.expand_matrix(
+            canonical_matrix, wires=self.wires, wire_order=wire_order
+        )
```

Comment:
do you also need to invert the matrix here if the op is inverted?

Reply:
I didn't do it because the original method didn't...

Reply:
oh lol!